python/mxnet/test_utils.py

```diff
@@ -86,13 +127,24 @@ def rand_sparse_ndarray(shape, storage_type, density=None):
         prod = np.prod(shape)
         num_cols = int(prod / shape[0])
         # sample index
-        idx_sample = rnd.rand(shape[0])
-        indices = np.argwhere(idx_sample < density).flatten()
+        if rsp_indices is not None:
+            indices = rsp_indices
+            assert(len(indices) <= shape[0])
+        else:
+            idx_sample = rnd.rand(shape[0])
+            indices = np.argwhere(idx_sample < density).flatten()
         if indices.shape[0] == 0:
-            result = mx.sparse_nd.zeros('row_sparse', shape)
+            result = mx.nd.zeros(shape, storage_type='row_sparse')
             return result, (np.array([]), np.array([], dtype='int32'))
         # generate random values
         val = rnd.rand(indices.shape[0], num_cols)
+
+        # Allow caller to override or adjust random values
+        if data_init is not None:
+            val.fill(data_init)
+        if modifier_func is not None:
+            val = assign_each(val, modifier_func)
+
```

Comment:
it' not possible for all situations, especially when modifier_func does an elemwise compare